subsys/bluetooth/services/dis.c

```diff
@@ -59,15 +107,134 @@ static ssize_t read_pnp_id(struct bt_conn *conn,
+			       BT_GATT_CHRC_READ, BT_GATT_PERM_READ,
+			       read_str, NULL, BT_GATT_DIS_FW_REV_STR_REF),
+#endif
+#if defined(CONFIG_BT_GATT_DIS_HW_REV)
+	BT_GATT_CHARACTERISTIC(BT_UUID_DIS_HARDWARE_REVISION,
+			       BT_GATT_CHRC_READ, BT_GATT_PERM_READ,
+			       read_str, NULL, BT_GATT_DIS_HW_REV_STR_REF),
+#endif
+#if defined(CONFIG_BT_GATT_DIS_SW_REV)
+	BT_GATT_CHARACTERISTIC(BT_UUID_DIS_SOFTWARE_REVISION,
+			       BT_GATT_CHRC_READ, BT_GATT_PERM_READ,
+			       read_str, NULL, BT_GATT_DIS_SW_REV_STR_REF),
+#endif
+
 );
+
+#if defined(CONFIG_BT_SETTINGS) && defined(CONFIG_BT_GATT_DIS_SETTINGS)
+static int dis_set(int argc, char **argv, size_t len_rd,
+		   settings_read_cb read_cb, void *store)
+{
+	int len;
+
+	if (!strcmp(argv[0], "manuf")) {
+		len = read_cb(store, &dis_manuf, sizeof(dis_manuf) - 1);
+		if (len < 0) {
+			BT_ERR("Failed to read manufacturer from storage"
+				       " (err %d)", len);
+		} else {
+			dis_manuf[len] = '\0';
+
+			BT_DBG("Manufacturer set to %s", dis_manuf);
+		}
+		return 0;
+	}
+	if (!strcmp(argv[0], "model")) {
+		len = read_cb(store, &dis_model, sizeof(dis_model) - 1);
+		if (len < 0) {
+			BT_ERR("Failed to read model from storage"
+				       " (err %d)", len);
+		} else {
+			dis_model[len] = '\0';
+
+			BT_DBG("Model set to %s", dis_model);
+		}
+		return 0;
+	}
+#if defined(CONFIG_BT_GATT_DIS_SERIAL_NUMBER)
+	if (!strcmp(argv[0], "serial")) {
+		len = read_cb(store, &dis_serial_number,
+			   sizeof(dis_serial_number) - 1);
+		if (len < 0) {
+			BT_ERR("Failed to read serial number from storage"
+				       " (err %d)", len);
+		} else {
+			dis_serial_number[len] = '\0';
+
+			BT_DBG("Serial number set to %s", dis_serial_number);
+		}
+		return 0;
+	}
+#endif
+#if defined(CONFIG_BT_GATT_DIS_FW_REV)
+	if (!strcmp(argv[0], "fw")) {
+		len = read_cb(store, &dis_fw_rev, sizeof(dis_fw_rev) - 1);
+		if (len < 0) {
+			BT_ERR("Failed to read firmware revision from storage"
+				       " (err %d)", len);
+		} else {
+			dis_fw_rev[len] = '\0';
+
+			BT_DBG("Firmware revision set to %s", dis_fw_rev);
+		}
+		return 0;
+	}
+#endif
+#if defined(CONFIG_BT_GATT_DIS_HW_REV)
+	if (!strcmp(argv[0], "hw")) {
+		len = read_cb(store, &dis_hw_rev, sizeof(dis_hw_rev) - 1);
+		if (len < 0) {
+			BT_ERR("Failed to read hardware revision from storage"
+				       " (err %d)", len);
+		} else {
+			dis_hw_rev[len] = '\0';
+
+			BT_DBG("Hardware revision set to %s", dis_hw_rev);
+		}
+		return 0;
+	}
+#endif
+#if defined(CONFIG_BT_GATT_DIS_SW_REV)
+	if (!strcmp(argv[0], "sw")) {
+		len = read_cb(store, &dis_sw_rev, sizeof(dis_sw_rev) - 1);
+		if (len < 0) {
+			BT_ERR("Failed to read software revision from storage"
+				       " (err %d)", len);
+		} else {
+			dis_sw_rev[len] = '\0';
+
+			BT_DBG("Software revision set to %s", dis_sw_rev);
+		}
+		return 0;
+	}
+#endif
+	return 0;
+}
+
+BT_SETTINGS_DEFINE(dis, dis_set, NULL, NULL);
+#endif /* CONFIG_BT_GATT_DIS_SETTINGS && CONFIG_BT_SETTINGS*/
+
+
```

Comment:
Two unnecessary empty lines at end of file, please remove

Reply:
OK